android/sdk/src/main/java/com/taobao/weex/ui/component/WXScroller.java

```diff
@@ -482,44 +485,45 @@ public void onScrollChanged(WXHorizontalScrollView scrollView, int x, int y, int
       scrollView.addView(mRealView, layoutParams);
       scrollView.setHorizontalScrollBarEnabled(false);
       mScrollerView = scrollView;
-        final WXScroller component = this;
-        final View.OnLayoutChangeListener listener = new View.OnLayoutChangeListener() {
-          @Override
-          public void onLayoutChange(View view, int left, int top, int right, int bottom, int oldLeft, int oldTop, int oldRight, int oldBottom) {
-            final View frameLayout = view;
-            scrollView.post(new Runnable() {
-              @Override
-              public void run() {
-                if (isLayoutRTL()) {
-                  int mw = frameLayout.getMeasuredWidth();
-                  scrollView.scrollTo(mw, component.getScrollY());
-                } else {
-                  boolean scrollToBegin = true;
-                  Object scrollToBeginOnLTR = getAttrs().get("scrollToBegin");
-                  if (scrollToBeginOnLTR instanceof String){
-                    if("false".equalsIgnoreCase((String)scrollToBeginOnLTR)){
-                      scrollToBegin = false;
-                    }
-                  }
-                  if (scrollToBegin){
-                    scrollView.scrollTo(0, component.getScrollY());
-                  }
+      final WXScroller component = this;
+      final View.OnLayoutChangeListener listener = new View.OnLayoutChangeListener() {
+        @Override
+        public void onLayoutChange(View view, final int left, int top, final int right, int bottom, final int oldLeft, int oldTop, final int oldRight, int oldBottom) {
+          final View frameLayout = view;
+          scrollView.post(new Runnable() {
+            @Override
+            public void run() {
+              if (mIslastDirectionRTL != null && isLayoutRTL() != mIslastDirectionRTL.booleanValue()) {
```

Comment:
As mIslastDirectionRTL is `Boolean`, which could cause null pointer exception here,

Why don't use `boolean mIslastDirectionRTL` directly?

Reply:
if use boolean, default is false, is not what i want.
I need record direction when last layout change occur, it not always false, and i need distinct is layout change trigger first or not.